vald/evm/evm.go

```diff
@@ -262,79 +264,49 @@ func (mgr Mgr) ProcessGatewayTxConfirmation(event *types.ConfirmGatewayTxStarted
-			continue
-		}
-
-		switch txlog.Topics[0] {
-		case ContractCallSig:
-			gatewayEvent, err := DecodeEventContractCall(txlog)
-			if err != nil {
-				mgr.logger().Debug(sdkerrors.Wrap(err, "decode event ContractCall failed").Error())
-				continue
-			}
+	events := mgr.processGatewayTxLogs(event.Chain, event.GatewayAddress, txReceipt.Logs)
+	mgr.logger().Infof("broadcasting vote %v for poll %s", events, event.PollID.String())
+	_, err = mgr.broadcaster.Broadcast(context.TODO(), voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain, events...)))
 
-			if err := gatewayEvent.ValidateBasic(); err != nil {
-				mgr.logger().Debug(sdkerrors.Wrap(err, "invalid event ContractCall").Error())
-				continue
-			}
+	return err
+}
 
-			events = append(events, types.Event{
-				Chain: event.Chain,
-				TxID:  event.TxID,
-				Index: uint64(i),
-				Event: &types.Event_ContractCall{
-					ContractCall: &gatewayEvent,
-				},
-			})
-		case ContractCallWithTokenSig:
-			gatewayEvent, err := DecodeEventContractCallWithToken(txlog)
-			if err != nil {
-				mgr.logger().Debug(sdkerrors.Wrap(err, "decode event ContractCallWithToken failed").Error())
-				continue
-			}
+// ProcessMultipleGatewayTxConfirmations votes on the correctness of an EVM chain multiple gateway transactions
+func (mgr Mgr) ProcessMultipleGatewayTxConfirmations(event *types.ConfirmGatewayTxsStarted) error {
+	if !slices.Any(event.Participants, func(v sdk.ValAddress) bool { return v.Equals(mgr.validator) }) {
+		f := slices.Map(event.PollMappings, func(m types.PollMapping) voteTypes2.PollID { return m.PollID })
+		mgr.logger("poll_ids", f).Debug("ignoring gateway txs confirmation poll: not a participant")
+		return nil
+	}
 
-			if err := gatewayEvent.ValidateBasic(); err != nil {
-				mgr.logger().Debug(sdkerrors.Wrap(err, "invalid event ContractCallWithToken").Error())
-				continue
-			}
+	txIDs := slices.Map(event.PollMappings, func(poll types.PollMapping) common.Hash { return common.Hash(poll.TxID) })
+	txReceipts, err := mgr.GetMultipleTxReceiptsIfFinalized(event.Chain, txIDs, event.ConfirmationHeight)
+	if err != nil {
+		return err
+	}
 
-			events = append(events, types.Event{
-				Chain: event.Chain,
-				TxID:  event.TxID,
-				Index: uint64(i),
-				Event: &types.Event_ContractCallWithToken{
-					ContractCallWithToken: &gatewayEvent,
-				},
-			})
-		case TokenSentSig:
-			gatewayEvent, err := DecodeEventTokenSent(txlog)
-			if err != nil {
-				mgr.logger().Debug(sdkerrors.Wrap(err, "decode event TokenSent failed").Error())
-			}
+	var votes []sdk.Msg
+	for i, result := range txReceipts {
+		pollID := event.PollMappings[i].PollID
+		txID := event.PollMappings[i].TxID
 
-			if err := gatewayEvent.ValidateBasic(); err != nil {
-				mgr.logger().Debug(sdkerrors.Wrap(err, "invalid event TokenSent").Error())
-				continue
-			}
+		keyvals := []interface{}{"chain", event.Chain, "poll_id", pollID.String(), "tx_id", txID.Hex()}
+		if result.Err() != nil {
```

Comment:
In `ProcessGatewayTxConfirmation`, we are only voting for empty when the receipt cannot be found, but here we are instead voing empty if the individal rpc call returns an error? That doesn't make sense to me.

Reply:
changed to only submit empty vote when tx not found and not finalized.